File: common-module/src/main/java/ru/kraser/technical_helper/common_module/dto/breakage/AppointBreakageExecutorDto.java
package ru.kraser.technical_helper.common_module.dto.breakage;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import ru.kraser.technical_helper.common_module.enums.Status;

import java.time.LocalDate;

public record AppointBreakageExecutorDto(
        @NotBlank(message = "Необходимо назначить исполнителя заявки на неисправность.")
        String executor,
        @NotNull(message = "Необходимо указать срок исполнений заявки на неисправность.")
        LocalDate deadline,
        @NotNull(message = "Заявка на неисправность должна иметь корректный статус.")
        Status status
) {
}

File: frontend/src/app/model/breakage/appoint-breakage-executor-dto.ts
import { Status } from "../../enum/status.enum";

export class AppointBreakageExecutorDto {
    executor: string;
    deadline: Date;
    status: Status;

    constructor(executor: string, deadline: Date, status: Status) {
        this.executor = executor;
        this.deadline = deadline;
        this.status = status;
    }
}